src/index.ts

```diff
@@ -291,10 +395,23 @@ const extension: JupyterFrontEndPlugin<void> = {
     window.console.table = (...args: any[]): void => {
       let data = '';
       args.forEach(arg => {
-        data +=
-          (typeof arg === 'object' && arg !== null
-            ? JSON.stringify(arg)
-            : arg) + ' ';
+        try {
+          data +=
+            (typeof arg === 'object' && arg !== null
+              ? JSON.stringify(arg)
+              : arg) + ' ';
+        } catch (e) {
+          try {
+            const msg =
+              'This error contains a object with a circular reference. During the process of removing the reference we could have removed duplicated attributes.\n';
+            const obj = JSON.stringify(arg, refReplacer);
+            cache = [];
+            console.error(msg, obj);
+            data += obj;
+          } catch (e) {
+            data += ' ';
+          }
```

Comment:
Maybe we could do a quick refactor if this piece of code is the same in all cases?

Reply:
ooh, thanks. I forgot about that.
I'm on it.